src/api/spec/controllers/webui/obs_factory/distributions_controller_spec.rb

```diff
@@ -0,0 +1,85 @@
+require 'rails_helper'
+require 'webmock/rspec'
+# WARNING: If you change owner tests make sure you uncomment this line
+# and start a test backend. Some of the Owner methods
+# require real backend answers for projects/packages.
+# CONFIG['global_write_through'] = true
+
+RSpec.describe Webui::ObsFactory::DistributionsController, type: :controller, vcr: true do
+  let(:factory) { create(:project, name: 'openSUSE:Factory') }
+  let!(:factory_staging_a) { create(:project, name: 'openSUSE:Factory:Staging:A', description: '') }
+  let!(:factory_ring_bootstrap) { create(:project, name: 'openSUSE:Factory:Rings:0-Bootstrap', description: '') }
+  let(:target_package) { create(:package, name: 'target_package', project: factory) }
+  let(:source_project) { create(:project, name: 'source_project') }
+  let(:source_package) { create(:package, name: 'source_package', project: source_project) }
+
+  describe 'GET #show' do
+    before do
+      allow(::ObsFactory::OpenqaJob).to receive(:find_all_by).and_return([:fake, :content])
+
+      get :show, params: { project: factory.name }
+    end
+
+    it { expect(response).to have_http_status(:success) }
+    it { expect(response).to render_template(:show) }
+
+    it { expect(assigns(:staging_projects).first.project).to eq(factory_staging_a) }
+
+    it { expect(assigns(:versions)[:source]).to eq(assigns(:distribution).source_version) }
+    it { expect(assigns(:versions)[:totest]).to eq(assigns(:distribution).totest_version) }
+    it { expect(assigns(:versions)[:published]).to eq(assigns(:distribution).published_version) }
+
+    it { expect(assigns(:ring_prjs).first.project).to eq(factory_ring_bootstrap) }
+    it { expect(assigns(:standard).nickname).to eq('standard') }
+    it { expect(assigns(:images).nickname).to eq('images') }
+    it { expect(assigns(:live)).to be_nil }
+    it { expect(assigns(:openqa_jobs)).to eq([:fake, :content]) }
+
+    context 'calculate_reviews' do
+      let!(:repo_checker_user) { create(:user, login: 'repo-checker') }
+      let!(:create_groups) do
```

Comment:
:thought_balloon: no need for `let!` as they are lazy evaluated in the `let!` below...

Reply:
create_groups is not called... but anyway, now it will be changed for merging it with the other loop